### PR TITLE
use artifact credentials controller

### PR DESCRIPTION
**Pull request by billiford**
- use artifact credentials controller where needed
- remove functions to get singular instances of a helm client
- change `url` reference in helm client definition to `repository` to align with .hal/config

### CODE REVIEW

File: pkg/http/core/artifacts_test.go [hunk 1]
```diff
@@ -55,6 +55,20 @@ var _ = Describe("Artifacts", func() {
...
 		})
 
+		When("getting the helm client returns an error", func() {
+			BeforeEach(func() {
+				fakeArtifactCredentialsController.HelmClientForAccountNameReturns(nil, errors.New("error getting helm client"))
+			})
+
+			It("returns an error", func() {
+				Expect(res.StatusCode).To(Equal(http.StatusBadRequest))
+				ce := getClouddriverError()
```
Comment (yasmin-abdullah) [inline pkg/http/core/artifacts_test.go:65]: Can you move this line to the BeforeEach block?

**Review by yasmin-abdullah — CHANGES_REQUESTED** [review]

File: pkg/http/core/artifacts_test.go (hunk 1 above)
Comment (billiford) [reply pkg/http/core/artifacts_test.go:65]: That func is reading from the http response, so I need to call that func after the http req has been made.

Comment (yasmin-abdullah) [reply pkg/http/core/artifacts_test.go:65]: That's fine, Would JustBeforeEach work in that case?

Comment (billiford) [reply pkg/http/core/artifacts_test.go:65]: Technically yes. The first jbe makes the request so internal jbe could read from the response, however I think that declaring a new variable scoped to the whole when block and adding a jbe to grab this is less readable and results in more lines. It's also more prone to user errors in the future.

If we decide to go this route of declaring and grabbing the clouddriver error in a jbe we should do this project-wide under another branch - but I think we should leave it as is for now.

Comment (yasmin-abdullah) [reply pkg/http/core/artifacts_test.go:65]: Agreed

**Review by yasmin-abdullah — APPROVED** [review]

**Pull request merged**